### PR TITLE
Fix unset and global commands to highlight all variable names

### PR DESCRIPTION
**Pull request by bitwisecook**
## Summary

- Fixed semantic token highlighting for `unset` and `global` commands to mark all variable name arguments as variables, not just the first one (issue #774)
- Implemented dynamic argument role resolution for `unset` to properly skip leading `-nocomplain` and `--` options before marking remaining arguments as `VarWrite`
- Added comprehensive tests to verify all variable names are highlighted correctly in both commands

### tcl-registry/src/commands/tcl/unset_.rs
- Added `unset_arg_roles()` function that dynamically determines which arguments are variable names by skipping leading options (`-nocomplain`, `--`)
- Changed `unset` command spec from static `arg_roles: &[(0, ArgRole::VarWrite)]` to use `arg_role_resolver: Some(unset_arg_roles)` for dynamic resolution

### tcl-lsp-core/src/semantic_tokens.rs
- Added `unset_marks_every_name_as_variable()` test verifying all three names in `unset x y z` highlight as variables
- Added test verifying `unset -nocomplain -- a b` correctly skips options and only marks `a` and `b` as variables
- Added `global_marks_every_name_as_variable()` test verifying all three names in `global a b c` highlight as variables

### tcl-registry/src/registry.rs
- Added `unset_marks_every_name()` test confirming `arg_indices_for_role()` returns all three argument indices for `unset x y z`
- Added `unset_skips_leading_options()` test confirming options are properly skipped and only real variable arguments are identified

## Validation

- Added 4 new unit tests covering the fixed behavior
- Tests verify both the basic case (all names highlighted) and the edge case (options properly skipped)
- Tests validate behavior at both the semantic tokens level and the registry level

https://claude.ai/code/session_01JrCk8CxaZz2vVGDRbc3bAh

### CODE REVIEW

File: rust/tcl-registry/src/commands/tcl/unset_.rs [hunk 1]
```diff
@@ -26,6 +26,24 @@ const FORMS: &[FormSpec] = &[FormSpec {
     synopsis: "unset ?-nocomplain? ?--? ?name name name ...?",
 }];
 
+/// `unset ?-nocomplain? ?--? ?name name name ...?` — every trailing word is a
+/// variable name, not just the first.  Resolve `VarWrite` dynamically (skipping
+/// the leading `-nocomplain` / `--` options, mirroring `lower_unset`) so all
+/// names highlight as variables rather than only the first (issue #774).
+fn unset_arg_roles(args: &[&str]) -> Vec<(u8, ArgRole)> {
+    let mut i = 0;
+    while i < args.len() && args[i].starts_with('-') {
```
[the remark this answers is not in the document]
Comment (bitwisecook) [reply rust/tcl-registry/src/commands/tcl/unset_.rs]: Good catch — fixed in 5deac974. Verified against tclsh 8.6/9.0 that `unset -foo bar` destroys both `-foo` and `bar` (and `unset -nocomplain -foo` unsets `-foo`), so `unset` recognises only `-nocomplain` and `--` as options; any other leading word is a variable name. The option scan now stops at the first word that is neither `-nocomplain` nor `--`, in both the arg-role resolver here and `lower_unset` (which shared the same bug), so semantic tokens and the analyser keep the `VarWrite`/def for dash-named targets. Added registry test coverage for `unset -foo bar` and `unset -nocomplain -foo`.

---
_Generated by [Claude Code](https://claude.ai/code)_
